Sources/NIOConcurrencyHelpers/NIOLock.swift

```diff
@@ -140,6 +140,12 @@ extension NIOLock {
         }
         return try body()
     }
+
+    // specialise Void return (for performance)
```

Comment:
Why is this more performant? This just calls down to `self.withLock(_:)`.

Reply:
I wouldn't know, i just copy pasted the exact function that `Lock` had, including the comment above it 😅

Reply:
@Lukasa should we remove this doc comment?

Reply:
It's a regular comment, not a doc comment. You can remove it if you'd like.

Reply:
The comment likely stopped being true when the method was marked `@inlinable`.

Reply:
Why don't we just suggest to use `withLock`? `withLockVoid` is pointless since about Swift 4.2 or so when `@inlinable` became available publicly.

Reply:
The honest answer is that I don’t think it’s worth the churn. While `withLockVoid` isn’t meaningfully better, plenty of existing code uses it and it makes the deprecation more painful than it needs to be.

Reply:
(Also it doesn’t totally seamlessly get replaced by `withLock`, due to single statement return value inference. That can cause you to need to explicitly discard the result, which this avoids.)